applications/ParticleMechanicsApplication/custom_utilities/mpm_boundary_rotation_utility.h

```diff
@@ -157,19 +157,19 @@ class MPMBoundaryRotationUtility: public CoordinateTransformationUtils<TLocalMat
 					array_1d<double,3> rN = rGeometry[itNode].FastGetSolutionStepValue(NORMAL);
 					this->Normalize(rN);
 
-					for( unsigned int i = 0; i < j; ++i)// Skip term (i,i)
-					{
-						rLocalMatrix(i,j) = 0.0;
-						rLocalMatrix(j,i) = 0.0;
-					}
-					for( unsigned int i = j+1; i < LocalSize; ++i)
-					{
-						rLocalMatrix(i,j) = 0.0;
-						rLocalMatrix(j,i) = 0.0;
-					}
-
+                    // Zero out row/column corresponding to normal displacement DoF except diagonal term (set to 1)
+                    // Applied IFF the local matrix passed is not empty [otherwise does nothing -- RHS only case]
+                    if (rLocalMatrix.size1() != 0) {
+                        for( unsigned int i = 0; i < LocalSize; ++i)
+                        {
+                            rLocalMatrix(i,j) = 0.0;
+                            rLocalMatrix(j,i) = 0.0;
+                        }
+                        rLocalMatrix(j, j) = 1.0; // set diagonal term to 1.0
```

Comment:
I think there is an identity matrix for this in boost

Reply:
I'm not too sure if the identity matrix will help in this case, since this operation is only for certain row/columns - the `j` index does not go through all DoFs, but only the first DoF for nodes flagged as `SLIP`:

```
for(unsigned int itNode = 0; itNode < rGeometry.PointsNumber(); ++itNode)
			{
				if(this->IsSlip(rGeometry[itNode]) )
				{
					// We fix the first displacement dof (normal component) for each rotated block
					unsigned int j = itNode * this->GetBlockSize();
```